Objects/stringlib/fastsearch.h

```diff
@@ -178,21 +178,30 @@ STRINGLIB(rfind_char)(const STRINGLIB_CHAR* s, Py_ssize_t n, STRINGLIB_CHAR ch)
 
 #undef MEMRCHR_CUT_OFF
 
-/* Change to a 1 to see logging comments walk through the algorithm. */
-#if 0 && STRINGLIB_SIZEOF_CHAR == 1
-# define LOG(...) printf(__VA_ARGS__)
-# define LOG_STRING(s, n) printf("\"%.*s\"", (int)(n), s)
-# define LOG_LINEUP() do {                                         \
-    LOG("> "); LOG_STRING(haystack, len_haystack); LOG("\n> ");    \
-    LOG("%*s",(int)(window_last - haystack + 1 - len_needle), ""); \
-    LOG_STRING(needle, len_needle); LOG("\n");                     \
-} while(0)
+
+/* Change to 1 or 2 to see logging comments walk through the algorithm.
+ * LOG_LEVEL == 1 print excludes input strings (useful for long inputs)
+ * LOG_LEVEL == 2 print includes input alignments */
+# define LOG_LEVEL 0
+#if LOG_LEVEL == 1 && STRINGLIB_SIZEOF_CHAR == 1
+    # define LOG(...) printf(__VA_ARGS__)
+    # define LOG_STRING(s, n)
+    # define LOG_LINEUP()
+#elif LOG_LEVEL == 2 && STRINGLIB_SIZEOF_CHAR == 1
+    # define LOG(...) printf(__VA_ARGS__)
+    # define LOG_STRING(s, n) printf("\"%.*s\"", (int)(n), s)
+    # define LOG_LINEUP() do {                                         \
+        LOG("> "); LOG_STRING(haystack, len_haystack); LOG("\n> ");    \
+        LOG("%*s",(int)(window_last - haystack + 1 - len_needle), ""); \
+        LOG_STRING(needle, len_needle); LOG("\n");                     \
+    } while(0)
 #else
-# define LOG(...)
-# define LOG_STRING(s, n)
-# define LOG_LINEUP()
+    # define LOG(...)
+    # define LOG_STRING(s, n)
+    # define LOG_LINEUP()
 #endif
 
+
```

Comment:
Revert unnecessary newlines. There are some more further down.

```suggestion
```

Reply:
Is it 1 line between functions in C? I sometimes find 1, sometimes 2.

Reply:
> Put blank lines around functions, structure definitions, and major sections inside functions.

I think 1, but leave them unchanged.

Reply:
Why?

Reply:
So how does it work then? If someone places a number of blank lines, then no-one can ever change it? What if it is stylistically the wrong number of lines?

Reply:
Currently style isn't enforced. So, it's better to not touch old code. I believe one day that will change, but until then no stylistic changes will be accepted.

Reply:
So under what circumstances is it going to change "1 day"? Maybe now is "(until) then"? How do I know?

Reply:
The lint job workflow will fail, making it impossible to merge your changes.

Reply:
I think it passed...

Reply:
I did work on around 70% of code in this module. Maybe that makes me eligible for adding an extra blank line?

This module was mostly using 2 lines. And it makes sense as it contains long functions. Those functions contain blank lines within them to separate code. Thus, separation of 2 lines makes it easier to navigate. In contrast where there are many short functions and there is no problem of needing to identify beginning of the next function while scrolling through many pages.

Having that said, if majority votes against, I have no problem removing it.

Reply:
> I think it passed...

When style is enforced, the workflow will fail if you don't follow the style guide. Currently it only checks for trailing space and a newline at the end of Python files.